L1Trigger/Phase2L1Taus/plugins/L1HPSPFTauProducer.cc

```diff
@@ -0,0 +1,319 @@
+#include "L1Trigger/Phase2L1Taus/plugins/L1HPSPFTauProducer.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include <cmath>  // std::fabs
+
+L1HPSPFTauProducer::L1HPSPFTauProducer(const edm::ParameterSet& cfg)
+    : moduleLabel_(cfg.getParameter<std::string>("@module_label")),
+      tauBuilder_(nullptr),
+      useChargedPFCandSeeds_(cfg.getParameter<bool>("useChargedPFCandSeeds")),
+      minSeedChargedPFCandPt_(cfg.getParameter<double>("minSeedChargedPFCandPt")),
+      maxSeedChargedPFCandEta_(cfg.getParameter<double>("maxSeedChargedPFCandEta")),
+      maxSeedChargedPFCandDz_(cfg.getParameter<double>("maxSeedChargedPFCandDz")),
+      useJetSeeds_(cfg.getParameter<bool>("useJetSeeds")),
+      minSeedJetPt_(cfg.getParameter<double>("minSeedJetPt")),
+      maxSeedJetEta_(cfg.getParameter<double>("maxSeedJetEta")),
+      minPFTauPt_(cfg.getParameter<double>("minPFTauPt")),
+      maxPFTauEta_(cfg.getParameter<double>("maxPFTauEta")),
+      minLeadChargedPFCandPt_(cfg.getParameter<double>("minLeadChargedPFCandPt")),
+      maxLeadChargedPFCandEta_(cfg.getParameter<double>("maxLeadChargedPFCandEta")),
+      maxLeadChargedPFCandDz_(cfg.getParameter<double>("maxLeadChargedPFCandDz")),
+      maxChargedIso_(cfg.getParameter<double>("maxChargedIso")),
+      maxChargedRelIso_(cfg.getParameter<double>("maxChargedRelIso")),
+      deltaRCleaning_(cfg.getParameter<double>("deltaRCleaning")),
+      applyPreselection_(cfg.getParameter<bool>("applyPreselection")),
+      debug_(cfg.getUntrackedParameter<bool>("debug", false)) {
+  if (debug_) {
+    std::cout << "<L1HPSPFTauProducer::L1HPSPFTauProducer (moduleLabel = " << moduleLabel_ << ")>:" << std::endl;
+  }
+
+  tauBuilder_.reset(new L1HPSPFTauBuilder(cfg));
```

Comment:
```suggestion
  tauBuilder_ = std::make_unique<L1HPSPFTauBuilder>(cfg);
```

Reply:
The result should be the same, but not having the raw `new` makes it clearer that the code is using a smart pointer instead of a raw pointer.